src/goto-symex/renaming_level.h

```diff
@@ -84,6 +86,63 @@ struct symex_level2t : public symex_renaming_levelt
 {
   symex_level2t() = default;
   ~symex_level2t() override = default;
+
+  /// In the case of a state split, we want to keep track of the overall
+  /// name generation across all states for identifier creation.
+  optionalt<current_namest> local_generations;
+
+  void erase(const irep_idt &identifier)
+  {
+    current_names.erase(identifier);
+    if (local_generations)
+      local_generations->erase(identifier);
+  }
+
+  /// If we have a local counter, use that instead of our global.
+  unsigned current_count(const irep_idt &identifier) const override
+  {
+    if(local_generations)
```

Comment:
This should be simpler -- the current count (i.e. the generation observed by a read) should always just be path-local-current-L2-names.at(identifier)